### PR TITLE
Add prettier to checkstyle workflow

### PR DESCRIPTION
**Pull request by bduffany**
Now that prettier is included in `buildfix.sh`, we can include it in checkstyle with less hassle when there are errors. Formats the following file types: js, jsx, ts, tsx, html, css, yaml, json, md

---

**Version bump**: TODO

### CODE REVIEW

File: .github/workflows/checkstyle.yaml [hunk 1]
```diff
@@ -44,6 +44,18 @@ jobs:
           echo "clang format errors:"
           cat clang-format-errors.txt
 
+      - name: prettier
+        # NOTE: Only run prettier on files that differ from master, since
+        # prettier can be slow.
+        run: |
+          git merge-base HEAD origin/master |
+            xargs git diff --name-only --diff-filter=AMRCT |
+            grep -P '\.(js|jsx|ts|tsx|html|css|yaml|json|md|xml)$' |
```
Comment (siggisim) [inline .github/workflows/checkstyle.yaml:53]: I'm sure there are a bunch of yaml, css, html, and md files that don't conform to prettier's style standard.

Should we fix those / see what they look like before enabling?

Comment (bduffany) [reply .github/workflows/checkstyle.yaml:53]: Sure -- https://github.com/buildbuddy-io/buildbuddy/pull/754

Comment (siggisim) [reply .github/workflows/checkstyle.yaml:53]: Seems reasonable if @tylerwilliams  is okay with it.

Comment (tylerwilliams) [reply .github/workflows/checkstyle.yaml]: 👍

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Pull request merged**